SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -414,10 +472,32 @@ void USpatialSender::SendRPC(TSharedRef<FPendingRPCParams> Params)
 		// Target object was destroyed before the RPC could be (re)sent
 		return;
 	}
-
 	UObject* TargetObject = Params->TargetObject.Get();
+
+	if (AActor* TargetActor = Cast<AActor>(TargetObject))
+	{
+		USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(PackageMap->GetEntityIdFromObject(TargetActor));
+		if (Channel == nullptr)
+		{
+			Channel = NetDriver->CreateActorChannel(TargetActor, NetDriver->GetSpatialOSNetConnection());
+		}
+		if (Channel != nullptr)
+		{
+			if (Channel->bCreatingNewEntity)
+			{
+				OutgoingOnCreateEntityRPCs.FindOrAdd(TargetActor).Add(Params);
+				return;
+			}
+		}
+		else
+		{
+			UE_LOG(LogSpatialSender, Warning, TEXT("Failed to create an Actor Channel for %s."), *TargetActor->GetName());
+		}
+	}
+
 	if (PackageMap->GetUnrealObjectRefFromObject(TargetObject) == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
 	{
+		// This is where we'll serialize this RPC and queue it to be added on entity creation
 		UE_LOG(LogSpatialSender, Verbose, TEXT("Trying to send RPC %s on unresolved Actor %s."), *Params->Function->GetName(), *TargetObject->GetName());
 		QueueOutgoingRPC(TargetObject, Params);
```

Comment:
Like I was saying earlier, I think you should investigate if this queuing is even needed any more. I don't think the TargetObject can be unresolved at this point any more.

Reply:
It could still potentially be unresolved for singletons in multi-worker scenario?

Reply:
We should leave a comment for the situations we think this could occur.

Reply:
Fair.

Reply:
I've added the comment @improbable-valentyn has suggested.